src/engraving/libmscore/paste.cpp

```diff
@@ -1129,9 +1164,12 @@ void Score::cmdPaste(const QMimeData* ms, MuseScoreView* view, Fraction scale)
                 qDebug("paste <%s>", data.data());
             }
             XmlReader e(data);
-            e.setPasteMode(true);
-            if (!pasteStaff(e, cr->segment(), cr->staffIdx(), scale)) {
-                return;
+            if (canPasteStaff(e, scale)) {
+                XmlReader e(data);
```

Comment:
XmlReader doesn't have a rewind...should add one perhaps?

Reply:
You already declared XmlReader e(data) at line 1166. We will get a warning here

Reply:
Declared? Rather called. So it might get called twice now.

Reply:
Correct,  you can't rewind it so you have to create a new one... happy to add a rewind function

Reply:
They're in different scopes so the behavior is well defined, but I agree stylistically some may prefer avoiding reusing the name (actually I don't like single letter variable names generally!).

Reply:
Took your suggestion of creating a separate method, creating a rewind method was a bit painful/dangerous due to inner workings of Qt.